analyses/ssd/flood_trigger/ssd_floodscan.md

```diff
@@ -0,0 +1,327 @@
+#            .rio.write_crs("EPSG:4326")
+#            .rio.clip(gdf_adm2["geometry"], all_touched=True))
+```
+
+```python
+# #somehow cannot save the file with these attributes
+# fs_clip.SFED_AREA.attrs.pop('grid_mapping')
+# fs_clip.NDT_SFED_AREA.attrs.pop('grid_mapping')
+# fs_clip.LWMASK_AREA.attrs.pop('grid_mapping')
+# fs_clip.to_netcdf(country_data_exploration_dir/'floodscan'/f'{iso3}_floodscan.nc')
+```
+
+```python
+fs_clip=xr.load_dataset(country_data_exploration_dir/'floodscan'/f'{iso3}_floodscan.nc')
+#I dont fully understand why, these grid mappings re-occur and what they mean
+#but if having them, later on getting
+fs_clip.SFED_AREA.attrs.pop('grid_mapping')
+fs_clip.NDT_SFED_AREA.attrs.pop('grid_mapping')
+fs_clip.LWMASK_AREA.attrs.pop('grid_mapping')
+fs_clip=fs_clip.rio.write_crs("EPSG:4326",inplace=True)
+```
+
+We plot the data. We can see that the resolution is high (300 arcseconds = 0.083 degrees). We therefore decide that when analyzing the data at the admin2 level, it is fine to only look at the cells with their centre within the admin2. 
+
+```python
+g=fs_clip.sel(time='2020-08-03T00:00:00.000000000').SFED_AREA.plot()
+gdf_adm2.boundary.plot(ax=g.axes);
+```
+
+there are three different variables in the data: SFED_AREA, NDT_SFED_AREA, and LWMASK_AREA. 
+SFED_AREA is the data we use. NDT_SFED_AREA is the same data but with no threshold, i.e. more senstitive. LWMASK_AREA is a mask with static water bodies. This is already not included in SFED_AREA. See for more information on the data types `floodscan.py` or the [user guide](https://www.aer.com/contentassets/22663ebfdb7c467599f363466b32770f/floodscan_data_users_guide_v05r01_r01.pdf)
+
+```python
+da_clip=fs_clip.SFED_AREA
+```
+
+We plot the values across the country. We focus on the rainy season, which is approximately from July till October with peak flooding months in August and September. 
+
+From the plot below we can see that the fraction of the area that is flooded (SFED_AREA) is generally very low, barely reacing above 0.2. I am not sure why this is the case. 
+Note that from the first graph you cannot see the higher values well. It does occur that there are values higher than 0.2 but much less often, see the second graph. 
+
+```python
+fs_clip_rainy=fs_clip.sel(time=fs_clip.time.dt.month.isin(range(7,11)))
+```
+
+```python
+fs_clip_rainy.hvplot.hist('SFED_AREA',alpha=0.5)
+```
+
+```python
+fs_clip_rainy.where(fs_clip_rainy>0.4).hvplot.hist('SFED_AREA',alpha=0.5)
+```
+
+When only focusing on the states of Jonglei and Unity, we can see that on average they show slightly higher values, though the majority of the values is still below 0.2
+
+```python
+(fs_clip_rainy.rio.write_crs("EPSG:4326").rio.clip(gdf_adm2.loc[gdf_adm2.ADM1_EN.isin(['Unity','Jonglei']),'geometry'])
+.hvplot.hist('SFED_AREA',alpha=0.5))
+```
+
+### Compute stats per county
+Now that we have analyzed the raster data, we can aggregate this to get the statistics per county. 
```

Comment:
I have to look more closely to figure out what you're calculating, but I need to go soon so before I do I just wanted to mention something. 

As I said before, I'm wondering if it's better to focus on total flooded area rather than the flood fraction of individual pixels. since I imagine there's probably quite some uncertainty at the pixel level. And even starting with looking at country-level rather than breaking it down into counties, which could bias the results. 

So what I mean is just for each day just summing the pixel fractions, and then dividing by the total number of pixels to get the amount of flooded area. From that you could identify when the major floods occur, and then see where they are happening geographically. Hope that makes sense.

Reply:
Hmm I like the thinking. However, rethinking it I think that is exactly what I am doing when taking the mean of the pixel fractions. 
mean(flood fraction of pixels within county) = sum(flood fraction of pixels within county)/#pixels within county. 
So actually instead of refering to the mean of the pixel fractions, it could refer to the total fraction of the county (or country) that is flooded. Do you agree or am I talking late afternoon bullshit here?

Reply:
> am I talking late afternoon bullshit here?

:joy: not at all, in fact you're totally right! 

Two more things that I was thinking about: 

1. Do you know how accurate the pixel fractions are? I haven't read any of the docs, but I'm wondering if pixel fraction is more either (a) a confidence on whether a given pixel is flooded or not, or (b) can sort of be ignored and we should just take any pixel > 0 as flooded
2. Probably out of scope but I'm wondering if we need to take into account clustering (spatial, but we could also do temporal to find events)

Reply:
1. As far as I understand is that the original resolution is 3 arcsec. At this resolution, the data is binary i.e. 1 is flooded 0 is non-flooded. Now the data we are using is 300 arcsec resolution. So I think (but this is not documented) that the fraction is just the fraction of the 1000 original pixels in our 1 pixel that are flooded. Does that make sense to you? Could double-check with the floodscan team
2. You mean if pixels close to each other show similair patterns?

Reply:
Yeah, was thinking the same thing about the clustering. Wondering if we have access to the `NDAYS` product described in the user documentation or if we can just produce this is ourselves?

![image](https://user-images.githubusercontent.com/8086547/153365314-d652f7c2-160e-4ab3-add0-aae165bb6354.png)

And interesting Tinka, definitely makes sense to me regarding the resolution/product.

Reply:
We don't have the NDAYS :( And maybe a bit harder to produce given we don't have the binary data, but we could do smth like NDAYS>pixel fraction 

For the clustering, makes sense!

Reply:
> As far as I understand is that the original resolution is 3 arcsec. At this resolution, the data is binary i.e. 1 is flooded 0 is non-flooded. Now the data we are using is 300 arcsec resolution. So I think (but this is not documented) that the fraction is just the fraction of the 1000 original pixels in our 1 pixel that are flooded. Does that make sense to you? Could double-check with the floodscan team

Good to know, thanks for explaining! 

> You mean if pixels close to each other show similair patterns?

Yeah, I would probably be inclined to define a "flood" as when many adjacent or nearly-adjacent pixels are > 1. If there are just random pixels > 1 with no connection to others then I would guess they are more likely to be FP.

Reply:
Yes! Added a note about it.